proposals/proposal_template.md

```diff
@@ -0,0 +1,140 @@
+
+### Known Consumers
+
+APIs should, ideally, have significant adoption once they are complete.
+Highlight any extensions that have committed to using this API.
+
+## Specification
+
+### Schema
+
+Include your API schema here (preferably with a TypeScript interface).  If you
+are adding new functionality or fields to an existing method, please be sure to
+call out explicitly what has changed.
+
+A few notes:
+
+*   Extension APIs are almost always asynchronous.  Anything that cannot be
+    resolved directly in the renderer must be asynchronous, and even if
+    something can be done in the renderer today, it should frequently be
+    asynchronous to prevent future breakage if we move something to the browser
+    process.
+*   Failures should be indicated by either synchronously throwing an error
+    (rare) or rejecting the returned promise.  (Note that in legacy versions,
+    this will populate
+    [chrome.runtime.lastError](https://developer.chrome.com/extensions/runtime#property-lastError).)
+    Do not provide another mechanism of indicating failure (e.g., don't accept
+    a failure callback).
+*   When practical, prefer accepting objects for input parameters.  This allows
+    us to expand APIs in the future in a non-breaking change by adding
+    additional optional properties.  Even if you currently only accept one
+    property, it often makes sense to wrap this in an object.
+
+### New Permissions
+
+| Permission Added | Suggested Warning |
+| ---------------- | ----------------- |
+| New Permission 1 | Add a proposed warning string for the permission.  If there should be no warning, provide justification as to why. Browser vendors can ultimately choose if there should be a warning and what it should be independently, but this can be useful to define especially if it has implications for the likelihood this proposal will succeed or be useful to developers. |
+
+Document any new permission added by this API.  Permissions are frequently the
+same as the API itself, e.g. the browser.storage API has the permission
+"storage".
+
+If this is a new API, it should add a new permission.  A new permission does
+_not_ always require a new permission warning (though it should frequently have
+one).  Adding a permission allows us to statically analyze extensions with
+greater accuracy, and avoid exposing unnecessary APIs.
+
+If this is a modification to an existing API, it _may_ require a new permission
+if the capabilities it adds are significantly different than the existing
+functionality in the API, and not already covered by the permission warning.
+Note that this is generally an indication that this might be better as a new
+API than a modification to an existing API.  If no new permission is needed,
+note why this falls within the current bounds of the API's capabilities.
+
+### Manifest File Changes
+
+Document if your API will require additional manifest fields to be added or
```

Comment:
Should we also encourage thoughts about validation/error handling? E.g. "If non-trivial, describe validation and error handling. When a failure is not critical, prefer a soft warning over a hard error. A hard error prevents an extension from loading, which adds friction to the adoption of potential future changes."